var/spack/repos/builtin/packages/autoconf/package.py

```diff
@@ -35,7 +35,9 @@ class Autoconf(Package):
     version('2.69', '82d05e03b93e45f5a39b828dc9c6c29b')
     version('2.62', '6c1f3b3734999035d77da5024aab4fbd')
 
-    depends_on('m4', type='build')
+    # Note: m4 is not a build-time dependency of autoconf. m4 is
+    # needed when autoconf runs, not when autoconf is built.
+    depends_on('m4')
```

Comment:
I'm essentially reverting a change that added a wrong `type="build"` description.

Are you sure `m4` is not needed when `autoconf` is built? Autoconf's documentation says that autoconf's configure script looks for it. Is it important to sort out this detail?

`m4` provides an executable that `autconf` needs to run. What kind of dependency is this? Is this even captured by `run`? I thought `run` is for shared libraries only. Do I need to add some additional setup to `m4` to make this work?

Also, there's an `m4 @1.4.6:` constraint that should be added.